src/lib.rs

```diff
@@ -13,8 +18,14 @@ pub fn init() {}
 
 // PTHREAD LAYER TO CALL LIBCTRU
 
-/// The main thread's thread ID. It is "null" because libctru didn't spawn it.
-const MAIN_THREAD_ID: libc::pthread_t = 0;
+#[derive(Clone, Debug)]
+#[repr(C)]
+struct PThread {
+    thread: ctru_sys::Thread,
+    os_thread: u32,
+}
+
+const_assert!(mem::size_of::<*mut PThread>() <= mem::size_of::<libc::pthread_t>());
```

Comment:
Maybe this is a pointless assertion, I left it in from trying to use `PThread` directly instead of a pointer.

Reply:
It's good to have since we do assume this relationship.